### PR TITLE
Implement device discovery within onvif-rs

### PR DESCRIPTION
**Pull request by DmitrySamoylov**
Story details: https://app.clubhouse.io/lumeo/story/202

### CODE REVIEW

File: src/discovery/mod.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+use async_std::stream::StreamExt;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+#[derive(Debug)]
+pub enum Error {
+    Network(std::io::Error),
+    Internal(String),
+}
+
+pub async fn discover(
```
Comment (chrisabruce) [inline src/discovery/mod.rs]: can you add comments here on how to use this function?

File: src/discovery/mod.rs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    duration: std::time::Duration,
+) -> Result<impl async_std::stream::Stream<Item = String>, Error> {
+    let probe = build_probe();
+    let probe_xml = yaserde::ser::to_string(&probe).map_err(Error::Internal)?;
+
+    let socket = (|| {
+        async {
+            let local_ipv4_addr = Ipv4Addr::UNSPECIFIED;
+            let multi_ipv4_addr = Ipv4Addr::new(239, 255, 255, 250);
```
Comment (chrisabruce) [inline src/discovery/mod.rs]: Lets define this as const for good measure.

File: src/discovery/mod.rs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            let local_socket_addr = SocketAddr::new(IpAddr::V4(local_ipv4_addr), 0);
+            let multi_socket_addr = SocketAddr::new(IpAddr::V4(multi_ipv4_addr), 3702);
```
Comment (chrisabruce) [inline src/discovery/mod.rs]: also this port should be const

**Review by chrisabruce — CHANGES_REQUESTED** [review]

**Review by victor-soloviev — APPROVED** [review]

File: src/discovery/mod.rs (hunk 1 above)
Comment (DmitrySamoylov) [reply src/discovery/mod.rs]: Sure. I've also changed its signature to make it more useful with `StreamExt` combinators.

**Review by chrisabruce — APPROVED** [review]

**Pull request merged**